### ISSUE TITLE
V2 Feature Request: add lifecycle interface support

### ISSUE THREAD

**Issue by kcmvp** [samber/do#61]
I have Java background , in springboot there is a interface named InitializingBean this interface give an opportunity to do some one-time initialization. We can introduce similar interface, service implement this interface will call the method automatically once after the service creation.

**Comment by samber** [comment]
I agree. I often add a small `svc.init()` before the end of my providers.

When would you call this method?

- calling `init()` after each service invocation
- calling all `init()` after a chain of service invocation

**Comment by kcmvp** [comment]
I think there is a bit difference between init() method and the case I describe. as go init() is package level and too many init method in a package is not a good practice(my opinion). my case is somelike "constructor" behavior, it should be called by the framework right after the service creation. not sure I address the case well or not.
